libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1597,6 +1604,69 @@ AP_GPS_UBLOX::_parse_gps(void)
 #endif
         break;
 
+#if AP_GPS_UBLOX_SATELITES_INFO_ENABLED
+    case MSG_SAT:
+    {
+        memset(state.satellites_prn,  0,  sizeof(state.satellites_prn));
+        memset(state.satellites_used,  0,  sizeof(state.satellites_used));
+        memset(state.satellites_snr,  0,  sizeof(state.satellites_snr));
+        memset(state.satellites_elevation,  0,  sizeof(state.satellites_elevation));
+        memset(state.satellites_azimuth,  0,  sizeof(state.satellites_azimuth));
+        state.satellites_visible = MIN(_buffer.sat.numSvs, UBLOX_NAV_SAT_MAX_SATELLITES);
+        for(int i = 0; i < state.satellites_visible; i++)
+        {
+            uint8_t ubx_sat_gnssId = _buffer.sat.sat_block[i].gnssId;
+            uint8_t ubx_sat_svId = _buffer.sat.sat_block[i].svId;
+            uint8_t svinfo_svid = 255;
+            switch (ubx_sat_gnssId) {
+            case 0:  // GPS: G1-G23 -> 1-32
+                if (ubx_sat_svId >= 1 && ubx_sat_svId <= 32) {
+                    svinfo_svid = ubx_sat_svId;
+                }
+                break;
+            case 1:  // SBAS: S120-S158 -> 120-158
+                if (ubx_sat_svId >= 120 && ubx_sat_svId <= 158) {
+                    svinfo_svid = ubx_sat_svId;
+                }
+                break;
+            case 2:  // Galileo: E1-E36 -> 211-246
+                if (ubx_sat_svId >= 1 && ubx_sat_svId <= 36) {
+                    svinfo_svid = ubx_sat_svId + 210;
+                }
+                break;
+            case 3:  // BeiDou: B1-B37 -> 159-163,33-64
+                if (ubx_sat_svId >= 1 && ubx_sat_svId <= 4) {
+                    svinfo_svid = ubx_sat_svId + 158;
+                } else if (ubx_sat_svId >= 5 && ubx_sat_svId <= 37) {
+                    svinfo_svid = ubx_sat_svId + 28;
+                }
+                break;
+            case 4:  // IMES: I1-I10 -> 173-182
+                if (ubx_sat_svId >= 1 && ubx_sat_svId <= 10) {
+                    svinfo_svid = ubx_sat_svId + 172;
+                }
+                break;
+            case 5:  // QZSS: Q1-A10 -> 193-202
+                if (ubx_sat_svId >= 1 && ubx_sat_svId <= 10) {
+                    svinfo_svid = ubx_sat_svId + 192;
+                }
+                break;
+            case 6:  // GLONASS: R1-R32 -> 65-96, R? -> 255
+                if (ubx_sat_svId >= 1 && ubx_sat_svId <= 32) {
+                    svinfo_svid = ubx_sat_svId + 64;
+                }
+                break;
+            }
+            state.satellites_used[i]       = static_cast<uint8_t>(_buffer.sat.sat_block[i].flags & 0x01);
```

Comment:
Again, no *relevant* constraint placed upon the value used to index into the array; need one.